src/dataExplorer/components/TagSelector.tsx

```diff
@@ -0,0 +1,172 @@
+  NewDataExplorerContext,
+} from 'src/dataExplorer/context/newDataExplorer'
+
+// Types
+import {RemoteDataState} from 'src/types'
+
+interface Prop {
+  loading: RemoteDataState
+  tagKey: string
+  tagValues: string[]
+  onSelectTagKey: (key: string) => void
+  onSelectTagValue: (value: string) => void
+}
+
+const TagValues: FC<Prop> = ({
+  loading,
+  tagKey,
+  tagValues,
+  onSelectTagKey,
+  onSelectTagValue,
+}) => {
+  const [valuesToShow, setValuesToShow] = useState([])
+  const [loadMore, setLoadMore] = useState(false)
+  const [index, setIndex] = useState(LOCAL_LIMIT)
+
+  useEffect(() => {
+    if (tagValues.length === 0) {
+      // Reset
+      setValuesToShow([])
+      setLoadMore(false)
+      setIndex(LOCAL_LIMIT)
+    } else {
+      const newValuesToShow = slice(tagValues, 0, index)
+      const newLoadMore = tagValues.length > LOCAL_LIMIT
+      setValuesToShow(newValuesToShow)
+      setLoadMore(newLoadMore)
+    }
+  }, [tagValues])
+
+  let list: JSX.Element | JSX.Element[] = []
+
+  if (loading === RemoteDataState.Error) {
+    list = (
+      <div className="tag-selector-value--list-item">
+        Failed to load tag values
+      </div>
+    )
+  } else if (
+    loading === RemoteDataState.Loading ||
+    loading === RemoteDataState.NotStarted
+  ) {
+    list = <WaitingText text="Loading tag values" />
+  } else if (loading === RemoteDataState.Done && tagValues.length) {
+    list = valuesToShow.map(value => (
+      <div
+        className="tag-selector-value--list-item"
+        key={value}
+        onClick={() => onSelectTagValue(value)}
+      >
+        {value}
+      </div>
+    ))
+  }
+
+  const handleLoadMore = () => {
+    const newIndex = index + LOCAL_LIMIT
+    if (loadMore) {
+      // Add more key values to show
+      const newValuesToShow = concat(
+        valuesToShow,
+        slice(tagValues, index, newIndex)
+      )
+      setValuesToShow(newValuesToShow)
+    }
+    const newLoadMore = newIndex < tagValues.length
+    setLoadMore(newLoadMore)
+    setIndex(newIndex)
+  }
+
+  const loadMoreButton = loadMore && (
+    <div className="load-more-button" onClick={handleLoadMore}>
+      + Load more
+    </div>
+  )
+
+  return (
+    <Accordion className="tag-selector-value">
+      <Accordion.AccordionHeader className="tag-selector-value--header">
+        <div onClick={() => onSelectTagKey(tagKey)}>
+          <SelectorTitle title={tagKey} />
+        </div>
+      </Accordion.AccordionHeader>
+      <div className="container-side-bar">
+        {list}
+        {loadMoreButton}
+      </div>
+    </Accordion>
+  )
+}
+
+const TagSelector: FC = () => {
+  const {tags, loadingTagKeys, loadingTagValues, selectTagKey} = useContext(
+    NewDataExplorerContext
+  )
+
+  const tagKeys: string[] = Object.keys(tags)
+
+  const handleSelectTagKey = (key: string) => {
+    if (tags[key].length === 0) {
+      // Only need to get tag values if currently no values
+      selectTagKey(key)
+    }
+  }
```

Comment:
Yes, here is the product requirement from #4510 

> Schema Browser will allow the one or more tag values to be selected

I think we will handle multiple tag values in `handleSelectTagValue` function. The function `selectTagKey` is used to fetch tag values on demand. Sorry the function name is not explicit on what it does and it is confusing 😅 I changed the name so that it's clearer